boards/arm/nucleo_f756zg/doc/nucleof756zg.rst

```diff
@@ -138,6 +140,10 @@ and a ST morpho connector. Board is configured as follows:
 - USB DP : PA12
 - I2C : PB8, PB9
 - PWM : PE13
+- SPI : PA4, PA5, PA6, PA7
+
+Note. The Arduino Uno v3 specified SPI device conflicts with the on-board ETH
```

Comment:
Interesting node. I wonder if there is a way to generate a warning if this is enabled (Using Kconfig maybe?)

Reply:
I did wonder initially about adding a #warning to the source to generate a compiler warning on compilation of pinmux.c if pins are conflicted.

I also considered kconfig dependency configuration but (for example) the spi_1 mosi signal can obviously be muxed to another pin for spi_1 to still work correctly with the on board Ethernet (although would break Arduino compatibility by doing so).

The documentation route seemed to be least messy/conflicting but happy to go with any suggestions you may have on how best to deal with this.

Reply:
Had no ideas on mind than the ones you proposed. 
Hence, it seems documentation is the best option we have, even if I'm certain people will face the issue before reading the doc.
Can you add comments as well in dts (before each conflicting node: arduino, ethernet) and pinmux? This will maximize the chances people read some info in this conflict.

Reply:
Absolutely, I'll get that sorted. When I get a chance I'll also stand up an equivalent PR for the f746 nucleo comments as it is also affected by this feature.